etl-data-pipeline/src/main/resources/db/routines/sp_provider.sql

```diff
@@ -0,0 +1,662 @@
+
+        IF OBJECT_ID('dbo.TMP_S_PROVIDER_POSTAL_LOCATOR', 'U') IS NOT NULL
+            drop table dbo.TMP_S_PROVIDER_POSTAL_LOCATOR;
+
+        IF OBJECT_ID('dbo.TMP_S_PROVIDER_TELE_LOCATOR_OFFICE', 'U') IS NOT NULL
+            drop table dbo.TMP_S_PROVIDER_TELE_LOCATOR_OFFICE;
+
+        IF OBJECT_ID('dbo.TMP_S_PROVIDER_TELE_LOCATOR_CELL', 'U') IS NOT NULL
+            drop table dbo.TMP_S_PROVIDER_TELE_LOCATOR_CELL;
+
+        IF OBJECT_ID('dbo.TMP_S_PROVIDER_LOCATOR', 'U') IS NOT NULL
+            drop table dbo.TMP_S_PROVIDER_LOCATOR;
+
+        IF OBJECT_ID('dbo.TMP_S_PROVIDER_QEC_ENTITY_ID', 'U') IS NOT NULL
+            drop table dbo.TMP_S_PROVIDER_QEC_ENTITY_ID;
+
+        IF OBJECT_ID('dbo.TMP_PRN_ENTITY_ID', 'U') IS NOT NULL
+            drop table dbo.TMP_PRN_ENTITY_ID;
+
+
+        IF OBJECT_ID('dbo.TMP_L_PROVIDER_N', 'U') IS NOT NULL
+            drop table dbo.TMP_L_PROVIDER_N;
+
+        IF OBJECT_ID('dbo.TMP_L_PROVIDER_E', 'U') IS NOT NULL
+            drop table dbo.TMP_L_PROVIDER_E;
+
+        IF OBJECT_ID('dbo.TMP_D_PROVIDER_N', 'U') IS NOT NULL
+            drop table dbo.TMP_D_PROVIDER_N;
+
+        IF OBJECT_ID('dbo.TMP_D_PROVIDER_E', 'U') IS NOT NULL
+            drop table dbo.TMP_D_PROVIDER_E;
+
+
+        COMMIT TRANSACTION;
+        select distinct [PROVIDER_UID]
+                      , [PROVIDER_LOCAL_ID]
+                      , [PROVIDER_RECORD_STATUS]
+                      , [PROVIDER_NAME_PREFIX]
+                      , [PROVIDER_FIRST_NAME]
+                      , [PROVIDER_MIDDLE_NAME]
+                      , [PROVIDER_LAST_NAME]
+                      , [PROVIDER_NAME_SUFFIX]
+                      , [PROVIDER_NAME_DEGREE]
+                      , [PROVIDER_GENERAL_COMMENTS]
+                      , [PROVIDER_QUICK_CODE]
+                      , [PROVIDER_REGISTRATION_NUM]
+                      , [PROVIDER_REGISRATION_NUM_AUTH]
+                      , [PROVIDER_STREET_ADDRESS_1]
+                      , [PROVIDER_STREET_ADDRESS_2]
+                      , [PROVIDER_CITY]
+                      , [PROVIDER_STATE]
+                      , [PROVIDER_STATE_CODE]
+                      , [PROVIDER_ZIP]
+                      , [PROVIDER_COUNTY]
+                      , [PROVIDER_COUNTY_CODE]
+                      , [PROVIDER_COUNTRY]
+                      , [PROVIDER_ADDRESS_COMMENTS]
+                      , [PROVIDER_PHONE_WORK]
+                      , [PROVIDER_PHONE_EXT_WORK]
+                      , [PROVIDER_EMAIL_WORK]
+                      , [PROVIDER_PHONE_COMMENTS]
+                      , [PROVIDER_PHONE_CELL]
+                      , [PROVIDER_ENTRY_METHOD]
+                      , [PROVIDER_LAST_CHANGE_TIME]
+                      , [PROVIDER_ADD_TIME]
+                      , [PROVIDER_ADDED_BY]
+                      , [PROVIDER_LAST_UPDATED_BY]
+        from dbo.TMP_S_PROVIDER_FINAL;
+
+        IF OBJECT_ID('dbo.TMP_S_PROVIDER_FINAL', 'U') IS NOT NULL
```

Comment:
Since the last statement is TMP_S_PROVIDER_FINAL, we might not get the select statement returned. Are there any issues with the return?

Reply:
Yeah we need to anticipate this scenario in the Java service..

Reply:
One approach we could take is to query from a session table. It would remove the need to remove the object after running the select. I current have the select as the last statement but if there is a way through the java service to handle running the drop statement that would be great!